src/Tokenizer/Token.php

```diff
@@ -183,10 +183,7 @@ public function equals($other, bool $caseSensitive = true): bool
             }
         }
 
-        // detect unknown keys
-        unset($otherPrototype[0], $otherPrototype[1]);
-
-        return empty($otherPrototype);
+        return true;
```

Comment:
can you tell a bit about why this change?

Reply:
The `@param` annotation ensures that when `$other` is an array, it only has key `0` and optionally key `1`. This means that at this point, `$otherPrototype` cannot have other keys than `0` and `1` so unsetting them always results in an empty array, making the function always return `true`.

But this is only true when using static analysis and this method is not `@internal` so I reverted the change and ignored the error instead.

Reply:
check :+1:   thanks for the details :)